agent/consul/health_endpoint.go

```diff
@@ -220,6 +220,8 @@ func (h *Health) ServiceNodes(args *structs.ServiceSpecificRequest, reply *struc
 	// If we're doing a connect or ingress query, we need read access to the service
 	// we're trying to find proxies for, so check that.
 	if args.Connect || args.Ingress {
+		// TODO: ACL Error improvements; can this be improved? What happens if we returned an error here?
+		// Is this similar to filters where we might want to return a hint?
 		if authz.ServiceRead(args.ServiceName, &authzContext) != acl.Allow {
```

Comment:
The `catalog` and `health` endpoints (and parts of `agent`) expect the filter style of ACL enforcement (200s instead of 403s).